src/_components/SubCategory/AddSubCategoriesContainer.jsx

```diff
@@ -0,0 +1,107 @@
+import * as React from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import _ from 'lodash';
+
+import AddSubCategoryComponent from '../../components/SubCategory/AddSubCategoryComponent';
+import { ToastMessage } from '../../_utils/ToastMessage';
+
+import { loadCategories } from '../../_actions/category.actions';
+import { createSubCategory } from '../../_actions/subcategory.actions';
+import resetToastMessageContent from '../../_actions/toastMessage.actions';
+
+class AddSubCategoriesContainer extends React.Component {
+  state = {
+    subCategory: '',
+    category: '',
+    saveButtonState: false
+  };
+
+  static getDerivedStateFromProps(nextProps) {
+    if (nextProps.toastMessageContent.type) {
+      if (nextProps.toastMessageContent.type === 'success') {
+        ToastMessage.success({
+          message: nextProps.toastMessageContent.message
+        });
+      } else if (nextProps.toastMessageContent.type === 'error') {
+        ToastMessage.error({
+          message: nextProps.toastMessageContent.message
+        });
+      }
+      nextProps.resetToastMessageContent();
+      nextProps.toggleModal();
+
+      return {
+        subCategory: '',
+        category: '',
+        saveButtonState: false
```

Comment:
@sachungo , the state is being updated here. The state of `saveButtonState` was true then becomes false when the modal is closed.

Reply:
I actually think I can handle that state change a bit better. Let me make some few alterations